molecule/test/unit/test_api.py

```diff
@@ -0,0 +1,34 @@
+from molecule.driver.base import Base
+from molecule.api import molecule_drivers
+
+
+def looks_iterable(a):
+    try:
+        (x for x in a)
+        return True
+    except TypeError:
+        return False
+
+
+def test_api_molecule_drivers():
+
+    # check if call without parameters returns a list of strings
+    results = molecule_drivers()
+    # print(results)
+    assert looks_iterable(results)
+    # delegated driver is expected to always be present
+    assert 'delegated' in results
+
+
+def test_api_molecule_drivers_as_dict():
+
+    # check that with as_dict it will return a dictionary of driver instances
+    results = molecule_drivers(as_dict=True)
+
+    assert looks_iterable(results)
```

Comment:
I would do the same as above and instead make explicit what type you expect.

Reply:
If you referred to checking for dict, I would prefer to avoid it because what I want is "hashable", maybe I would even change the signature to make this clear. I may not return a dictionary, but something that behaves like one in the future.

Reply:
Shameless copy/pasta:

> my point is: test what you expect the code to return
> ↳if you plan to change it in the future, then change the test
> ↳you might not come back to this test for $reason and then it could make no sense to someone else
> ↳what is "looks iterable" when there is some random failure a few months down the line
> ↳better practice to test against explicit types

Reply:
I run out of energy around this patch and that's bad because the driver loading code in master is currently very broken, so broken that any pip conflict would break molecule, not even `molecule --version` would work. Here is the proof https://gist.github.com/ssbarnea/526c2c453d2714e659a9e1f6de1fbf56#file-gistfile1-txt-L61-L85

I more than open to improve the testing and the behavior of the role so anyone is welcomed to update this CR and bring up to mergeable standards. As long it no longer contains `iter_entry_points` I will support it.